torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -48,6 +51,221 @@ c10::FunctionSchema getExecuteSchema() {
+                              const Module& orig_module,
+                              const c10::Dict<IValue, IValue>& method_compile_spec,
+                              c10::DictTypePtr any_dict_ty) {
+  const c10::QualifiedName qual_backend_name(
+      {"__torch__",
+       "torch",
+       "classes",
+       detail::kBackendsNamespace,
+       backend_name});
+  // TODO: Validate method_compile_spec.
+
+  // Clone orig_module to make sure backend transformation is
+  // functional.
+  auto cloned_module = orig_module.clone();
+
+  // Generate LoweredModule.
+  Module loweredModule(
+      "torch.jit." + backend_name + "LoweredModule",
+      std::make_shared<CompilationUnit>(),
+      /*shouldMangle=*/true);
+
+  // Generate attributes.
+  // This is the original cloned and preprocessed module.
+  loweredModule.register_attribute(
+      "__processed_module",
+      AnyType::get(),
+      cloned_module._ivalue(),
+      /*is_param=*/false);
+
+  // This is for the method_compile_spec passed in to to_<backend> or
+  // loaded from an exported model.
+  loweredModule.register_attribute(
+      "__method_compile_spec",
+      any_dict_ty,
+      method_compile_spec,
+      /*is_param=*/false);
+
+  // This is a pointer to a backend instance that is used to access
+  // compile and execute functions.
+  auto cls = getCustomClass(qual_backend_name.qualifiedName());
+  TORCH_INTERNAL_ASSERT(cls);
+  c10::intrusive_ptr<torch::CustomClassHolder> backend;
+  loweredModule.register_attribute(
+      "__backend", cls, IValue::make_capsule(backend));
+
+  // This is the list of opaque backend handles returned by
+  // backend.compile.
+  loweredModule.register_attribute(
+      "__handles",
+      any_dict_ty,
+      c10::impl::GenericDict(
+          any_dict_ty->getKeyType(), any_dict_ty->getValueType()),
+      /*is_param=*/false);
+
+  // Methods.
+
+  // This is a helper function for creating a new instance of the
+  // backend class.
+  static const auto create_backend_ct = CodeTemplate(R"(
+            def __create_backend(self):
+                self.__backend = $name()
+            )");
+  TemplateEnv create_backend_te;
+  create_backend_te.s("name", qual_backend_name.qualifiedName());
+  loweredModule.define(
+      create_backend_ct.format(create_backend_te), loweredModuleResolver());
+
+  // getstate and setstate are for serialization/deserialization of
+  // the LoweredModule.
+  loweredModule.define(
+      R"(
+            def __getstate__(self):
+                return self.__method_compile_spec, self.__processed_module
+            )",
+      loweredModuleResolver());
+
+  loweredModule.define(
+      R"(
+            def __setstate__(self, state):
+                self.__method_compile_spec = state[0]
+                self.__processed_module = state[1]
+                self.__create_backend()
+                self.__handles = self.__backend.compile(self.__processed_module, self.__method_compile_spec)
```

Comment:
So at the moment we will do compilation at load time? But we do need to be able to this AOT right? And if so I am wondering how are we going to store compiled methods, or whatever else the processed_module needs to store, that does not have native serialization support in pytorch. Do we cast it to std::vector<uint8_t> or something like that?

Reply:
Sorry for some reason I missed this comment. Probably it's an experimental PR and I was focused on the "formal" ones. You might already got your answers from #51232